x-pack/plugins/enterprise_search/common/connectors/native_connectors.ts

```diff
@@ -146,6 +146,260 @@ export const NATIVE_CONNECTOR_DEFINITIONS: Record<string, NativeConnector | unde
+        value: '',
+      },
+      ssl_enabled: {
+        default_value: false,
+        depends_on: [],
+        display: DisplayType.TOGGLE,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.mssql.configuration.sslEnabledLabel',
+          {
+            defaultMessage: 'Enable SSL',
+          }
+        ),
+        options: [],
+        order: 7,
+        required: true,
+        sensitive: false,
+        tooltip: '',
+        type: FieldType.BOOLEAN,
+        ui_restrictions: [],
+        validations: [],
+        value: false,
+      },
+      ssl_ca: {
+        default_value: '',
+        depends_on: [{ field: 'ssl_enabled', value: true }],
+        display: DisplayType.TEXTBOX,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.mssql.configuration.sslCertificateLabel',
+          {
+            defaultMessage: 'SSL certificate',
+          }
+        ),
+        options: [],
+        order: 8,
+        required: true,
+        sensitive: false,
+        tooltip: '',
+        type: FieldType.STRING,
+        ui_restrictions: [],
+        validations: [],
+        value: '',
+      },
+      schema: {
+        default_value: '',
+        depends_on: [],
+        display: DisplayType.TEXTBOX,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.mssql.configuration.schemaLabel',
+          {
+            defaultMessage: 'Schema',
+          }
+        ),
+        options: [],
+        order: 9,
+        required: false,
+        sensitive: false,
+        tooltip: '',
+        type: FieldType.STRING,
+        ui_restrictions: [],
+        validations: [],
+        value: '',
+      },
+      fetch_size: {
+        default_value: 50,
+        depends_on: [],
+        display: DisplayType.NUMERIC,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.mssql.configuration.rowsFetchedLabel',
+          {
+            defaultMessage: 'Rows fetched per request',
+          }
+        ),
+        options: [],
+        order: 10,
+        required: false,
+        sensitive: false,
+        tooltip: '',
+        type: FieldType.INTEGER,
+        ui_restrictions: ['advanced'],
+        validations: [],
+        value: 50,
+      },
+      retry_count: {
+        default_value: 3,
+        depends_on: [],
+        display: DisplayType.NUMERIC,
+        label: i18n.translate(
+          'xpack.enterpriseSearch.nativeConnectors.mssql.configuration.retriesLabel',
+          {
+            defaultMessage: 'Retries per request',
+          }
+        ),
+        options: [],
+        order: 11,
+        required: false,
+        sensitive: false,
+        tooltip: '',
+        type: FieldType.INTEGER,
+        ui_restrictions: ['advanced'],
+        validations: [],
+        value: 3,
+      },
+      validate_host: {
+        default_value: false,
+        depends_on: [],
+        display: DisplayType.TOGGLE,
```

Comment:
display: "text" in the backend and seems like a bug on our side! cc @navarone-feekery